### ISSUE TITLE
Failure when pulling a vault twice

### ISSUE THREAD

**Issue by tegefaulkes** [MatrixAI/Polykey-CLI#196]
### Describe the bug

We encountered a crash when testing. It seems to be triggered by pulling a node twice.

```
~/Downloads took 16s 
❯ ./pk vaults pull rippling vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
ErrorPolykeyRemote: Remote error from RPC call
  localHost     ::1
  localPort     39264
  remoteHost    ::1
  remotePort    42179
  command       vaultsPull
  timestamp     Tue Jun 11 2024 13:52:39 GMT+1000 (Australian Eastern Standard Time)
  cause: TypeError: Cannot read properties of undefined (reading 'length')
```

```
❯ ./dist/polykey.js vaults pull rippling vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
ErrorPolykeyRemote: Remote error from RPC call
  localHost    ::1
  localPort    41060
  remoteHost    ::1
  remotePort    39037
  command    vaultsPull
  timestamp    Tue Jun 11 2024 14:12:41 GMT+1000 (Australian Eastern Standard Time)
  cause: TypeError: Cannot read properties of undefined (reading 'length')
```

Theses are all the details I have.

### To Reproduce

1. Clone a vault.
2. Pull the vault that was just cloned
3. Pull the vault again, it should crash.

### Expected behavior

Pulling a vault should not crash

### Platform (please complete the following information)
 - Version Polykey `0.4.1`

### Notify maintainers

@tegefaulkes

**Comment by CMCDragonkai** [comment]
The node doesn't crash, it's just that pulling the vault twice doesn't work, I put new changes into the vault to be pulled.

**Comment by CMCDragonkai** [comment]
The node is crashing randomly due to another reason `ErrorQUICConnectionIdleTimeout`.

**Comment by tegefaulkes** [comment]
There is still a history discrepancy when cloning as well.

```
❯ ./dist/polykey.js vaults log rippling
commitId     cf16bae7aa6fc63267d9584d960e4516969b5fe7
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:26:38 GMT+1000 (Australian Eastern Standard Time)
message      "Merge branch \'HEAD\' of http://\n"
commitId     1b2b3ae2abb48e1cd8314071c6acc3ea369894fa
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:24:54 GMT+1000 (Australian Eastern Standard Time)
message      "Super.md added\n"
commitId     6cf8635b3dc0f3c678010cdc2f1172c5b8077a8a
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:09:07 GMT+1000 (Australian Eastern Standard Time)
message      "Merge branch \'HEAD\' of http://\n"
commitId     99b08b870487a5d5ebf0f415c75fb1be3e7ebcae
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:08:52 GMT+1000 (Australian Eastern Standard Time)
message      "CompanyRegistrationCertificate.pdf added\n"
commitId     53abaade1d1c8816e9ab7d6290e8aa320e78b0a6
committer    vet1dhoduhkmn4749r8usiopivklr8i4qoh9kjnhtgeg5rie4nvh0
timestamp    Tue Jun 11 2024 13:04:33 GMT+1000 (Australian Eastern Standard Time)
message      "Initial Commit\n"
commitId     5b99546d0cd9bffb7e536440b70e7b3d24af70f9
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 12:51:21 GMT+1000 (Australian Eastern Standard Time)
message      "BusinessDetails.md added\n"
commitId     f3e1b663e91bf9f5105ae03f593a081338662a25
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 12:47:57 GMT+1000 (Australian Eastern Standard Time)
message      "Initial Commit\n"
```

```
cmcdragonkai ➜ matrix-framework-13-ryzen-7040  ➜ ~/Projects/matrix-employees/documents/Aryan Jassal
 $ ~/Projects/Polykey-CLI/dist/polykey.js vaults log rippling
commitId     4dc9ae9e0f801334e9750bb8bd09098d3088aa62
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:50:51 GMT+1000 (Australian Eastern Standard Time)
message      "TaxWithholdingDeclaration.pdf added\n"
commitId     9731f91086e93b0a5f6ff853f2193536335001e1
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:50:29 GMT+1000 (Australian Eastern Standard Time)
message      "MedicareLevyVariation.pdf added\n"
commitId     a790557f6936b1155cb73511173704b0b5ebb70c
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:50:06 GMT+1000 (Australian Eastern Standard Time)
message      "SuperForm.pdf added\n"
commitId     812f92b56e81f56d23d04aa6335825526da17afa
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:49:37 GMT+1000 (Australian Eastern Standard Time)
message      "CasualEmploymentContract.pdf added\n"
commitId     1b2b3ae2abb48e1cd8314071c6acc3ea369894fa
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:24:54 GMT+1000 (Australian Eastern Standard Time)
message      "Super.md added\n"
commitId     99b08b870487a5d5ebf0f415c75fb1be3e7ebcae
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 13:08:52 GMT+1000 (Australian Eastern Standard Time)
message      "CompanyRegistrationCertificate.pdf added\n"
commitId     5b99546d0cd9bffb7e536440b70e7b3d24af70f9
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 12:51:21 GMT+1000 (Australian Eastern Standard Time)
message      "BusinessDetails.md added\n"
commitId     f3e1b663e91bf9f5105ae03f593a081338662a25
committer    vcj3ufun2506jn44q7rr9bc1hv4c6eebicusfj1bl9sr27ookknlg
timestamp    Tue Jun 11 2024 12:47:57 GMT+1000 (Australian Eastern Standard Time)
message      "Initial Commit\n"
```

**Comment by tegefaulkes** [comment]
There is already a test checking for both pulling multiple times and having a proper commit history afterwards. These changes were made in `Polykey` `1.5.0` So I'm not sure how we're getting such bad behaviour in the CLI testing.

I need to make sure that both @CMCDragonkai and @aryanjassal were using `Polykey-CLI` version `0.4.1` while testing this. Right now I think one of you using version 0.4.0 or older would be the most likely cause.

**Comment by tegefaulkes** [comment]
I tried to re-create the problem with @aryanjassal just now. We did the following.

1. created a vault
2. added a secret
3. cloned
4. added a secret
5. pulled
6. modified a secret
7. added a 3rd secret
8. pulled.

After every pull I checked the log and the contents. There were *no* problems found when doing this.

Were trying an old version to see if we can recreate the problem.

**Comment by tegefaulkes** [comment]
Attempted test 2 with @aryanjassal running `Polykey-CLI` `0.3.1`. We got the same results exactly with no issue.

We tested the same in reverse where @aryanjassal was pulling with the old version. We could re-create the problem with the logs. This makes sense since that version is older than the log fixes anyway.

During this testing we could net re-create the issue with pulling failing. So I'm concluding that this was just an issue with using an old incompatible version.
